nano/node/socket.cpp

```diff
@@ -72,69 +67,39 @@ void nano::socket::async_read (std::shared_ptr<std::vector<uint8_t>> buffer_a, s
 	}
 }
 
-void nano::socket::async_write (nano::shared_const_buffer const & buffer_a, std::function<void(boost::system::error_code const &, size_t)> const & callback_a, nano::buffer_drop_policy drop_policy_a)
+void nano::socket::async_write (nano::shared_const_buffer const & buffer_a, std::function<void(boost::system::error_code const &, size_t)> const & callback_a)
 {
 	auto this_l (shared_from_this ());
 	if (!closed)
 	{
-		if (writer_concurrency == nano::socket::concurrency::multi_writer)
-		{
-			boost::asio::post (strand, boost::asio::bind_executor (strand, [buffer_a, callback_a, this_l, drop_policy_a]() {
-				if (!this_l->closed)
-				{
-					bool write_in_progress = !this_l->send_queue.empty ();
-					auto queue_size = this_l->send_queue.size ();
-					if (queue_size < this_l->queue_size_max || (drop_policy_a == nano::buffer_drop_policy::no_socket_drop && queue_size < (this_l->queue_size_max * 2)))
-					{
-						this_l->send_queue.emplace_back (nano::socket::queue_item{ buffer_a, callback_a });
-					}
-					else if (auto node_l = this_l->node.lock ())
+		++queue_size;
+		boost::asio::post (strand, boost::asio::bind_executor (strand, [buffer_a, callback_a, this_l]() {
+			if (!this_l->closed)
+			{
+				this_l->start_timer ();
```

Comment:
If deadline is also atomic, is it required to be in strand?

Reply:
It seems like there are other things to clean up before we can remove the strand entirely.